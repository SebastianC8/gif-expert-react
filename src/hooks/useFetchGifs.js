import { useEffect, useState } from "react";
import { getGifs } from '../helpers/getGifs';

/**
 * Custom hook
 */
export const useFetchGifs = (category) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    });

    /**
    * Hace que la funciÃ³n solo se ejecute la primera vez que se renderiza el componente
    * @param Callback
    * @param Array
    */
    useEffect(() => getGifs(category).then((img) => setState({
        data: img,
        loading: false
    })), [category]);

    return state;

}