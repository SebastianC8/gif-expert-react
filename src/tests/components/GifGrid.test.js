import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

/**
 * Simula petición a este archivo
 */
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas componente <GifGrid/>', () => {
    
    test('Componente ok', () => {
       
        /**
         * Formato de valor esperado en el retorno.
         */
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category = { 'Inuyasha' } />);
        expect(wrapper).toMatchSnapshot();

    });

    test('Mostrar items cuando se cargan imágenes con useFetchGifs', () => {
 
        /**
         * Objeto esperado
         */
        const gifs = [{ id: 1, url: 'https://localhost/path', title: 'Título imagen' }];

        /**
         * Simula retorno de data esperada
         */
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category = { 'Inuyasha' } />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    });
    
});
