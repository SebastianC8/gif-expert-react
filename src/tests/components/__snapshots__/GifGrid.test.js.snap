// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente <GifGrid/> Componente ok 1`] = `
<Fragment>
  <h3>
    Inuyasha
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  >
    <ol />
  </div>
</Fragment>
`;

exports[`Pruebas componente <GifGrid/> Mostrar items cuando se cargan imágenes con useFetchGifs 1`] = `
<Fragment>
  <h3>
    Inuyasha
  </h3>
  <div
    className="card-grid"
  >
    <ol>
      <GifGridItem
        id={1}
        key="1"
        title="Título imagen"
        url="https://localhost/path"
      />
    </ol>
  </div>
</Fragment>
`;
